common-npm-packages/docker-common/containerimageutils.ts

```diff
@@ -253,3 +253,58 @@ export function getImageIdFromBuildOutput(output: string): string {
 
     return "";
 }
+export function getBaseImageDigestDockerFile(dockerFileContentPath: string): string {
+    // This method checks if there is FROM image@sha256:digest present in Dockerfile
+    // if matched it returns digest
+    // if not, it returns null
+    
+    try {
+        var dockerFileContent=fs.readFileSync(dockerFileContentPath).toString();
+        if (!dockerFileContent || dockerFileContent == "") {
+            return null;
+        }        
+        var lines = dockerFileContent.split(/[\r?\n]/);
+        var aliasToImageNameMapping: Map<string, string> = new Map<string, string>();
+        var baseImage = "";
+    
+        // Added regex pattern to check line starts with FROM IMAGE
+        const matchPatternForDockerImage = new RegExp(/^FROM\s+IMAGE/); 
+
+        for (var i = 0; i < lines.length; i++) {
+            const currentLine = lines[i].trim();
+            
+            if (!currentLine.toUpperCase().match(matchPatternForDockerImage)) {
+                continue;
+            }
+            var nameComponents = currentLine.substring(4).toLowerCase().split(" as ");
+            var prospectImageName = nameComponents[0].trim();
+    
+            if (nameComponents.length > 1) {
+                var alias = nameComponents[1].trim();
+    
+                if (aliasToImageNameMapping.has(prospectImageName)) {
+                    aliasToImageNameMapping.set(alias, aliasToImageNameMapping.get(prospectImageName));
+                } else {
+                    aliasToImageNameMapping.set(alias, prospectImageName);
+                }
+    
+                baseImage = aliasToImageNameMapping.get(alias);
+            } else {
+                baseImage = aliasToImageNameMapping.has(prospectImageName)
+                    ? aliasToImageNameMapping.get(prospectImageName)
+                    : prospectImageName;
+            }
+        }
+        // check below line with array length check
+        if(baseImage.split('@').length > 1){
+            if (baseImage.split('@')[1].split(':').length > 1) {
```

Comment:
Please reuse calculation results from the previous line so you don't need to make the same split() call again